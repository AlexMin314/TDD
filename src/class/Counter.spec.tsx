import React from 'react';
import { shallow } from 'enzyme';
import Counter from './Counter';
import { render, fireEvent } from '@testing-library/react';

describe('<Counter />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Counter />);
    expect(wrapper).toMatchSnapshot();
  });

  // state
  it('has initial number', () => {
    const wrapper: any = shallow(<Counter />);
    expect(wrapper.state().number).toBe(0);
  });
  it('increases', () => {
    const wrapper: any = shallow(<Counter />);
    wrapper.instance().handleIncrease();
    expect(wrapper.state().number).toBe(1);
  });
  it('decreases', () => {
    const wrapper: any = shallow(<Counter />);
    wrapper.instance().handleDecrease();
    expect(wrapper.state().number).toBe(-1);
  });

  // events
  it('calls handleIncrease', () => {
    const wrapper: any = shallow(<Counter />);
    const plusButton = wrapper.findWhere(
      (node: any) => node.type() === 'button' && node.text() === '+1'
    );
    plusButton.simulate('click');
    expect(wrapper.state().number).toBe(1);
  });
  it('calls handleDecrease', () => {
    const wrapper: any = shallow(<Counter />);
    const minusButton = wrapper.findWhere(
      (node: any) => node.type() === 'button' && node.text() === '-1'
    );
    minusButton.simulate('click');
    const number = wrapper.find('h2');
    expect(number.text()).toBe('-1');
  });
});

describe('<Counter />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Counter />);
    expect(container).toMatchSnapshot();
  });

  it('has initial number', () => {
    const { getByText } = render(<Counter />);
    getByText('0');
  });
  it('increases', () => {
    const { getByText } = render(<Counter />);
    const stateNumber = getByText('0');
    const increaseButton = getByText('+1');

    fireEvent.click(increaseButton);
    fireEvent.click(increaseButton);

    expect(stateNumber).toHaveTextContent('2');
    expect(stateNumber.textContent).toBe('2');
  });
  it('decreases', () => {
    const { getByText } = render(<Counter />);
    const stateNumber = getByText('0');
    const decreaseButton = getByText('-1');

    fireEvent.click(decreaseButton);
    fireEvent.click(decreaseButton);

    expect(stateNumber).toHaveTextContent('-2');
    expect(stateNumber.textContent).toBe('-2');
  });
});