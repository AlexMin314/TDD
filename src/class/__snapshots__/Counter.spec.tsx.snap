// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches snapshot 1`] = `
<div>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;

exports[`<Counter /> matches snapshot 2`] = `
<div>
  <div>
    <h2>
      0
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
  </div>
</div>
`;
